net/shorewall/Makefile

```diff
@@ -0,0 +1,117 @@
+PKG_LICENSE:=GPL-2.0+
+PKG_LICENSE_FILES:=COPYING
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/shorewall
+    SECTION:=net
+    CATEGORY:=Network
+    DEPENDS:=+ip +iptables +kmod-ipt-hashlimit +kmod-ipt-raw +iptables-mod-hashlimit +shorewall-core +perl +perlbase-autoloader +perlbase-autouse +perlbase-dynaloader +perlbase-digest +perlbase-findbin +perlbase-getopt +perlbase-hash
+    TITLE:=Shorewall Central Administration System
+    URL:=http://www.shorewall.net/
+    SUBMENU:=Firewall
+endef
+
+define Package/shorewall/description
+	The Shoreline Firewall, is high-level tool for configuring Netfilter.
+
+	Shorewall allows for central administration of multiple IPv4 firewalls.
+	This is the full Shorewall product which will compile Shorewall scripts.
+	It is not recommended to run it on a low memory system.
+
+	Note: This is the IPv4 implementation of Shorewall.
+	      This full Shorewal packages also installs Perl which can make the image big (about +2M).
+endef
+
+CONFIGURE_ARGS += \
+	vendor=openwrt
+
+define Build/Compile
+	DESTDIR=$(PKG_INSTALL_DIR) $(PKG_BUILD_DIR)/install.sh
+endef
+
+define Package/shorewall/conffiles
+/etc/shorewall/accounting
+/etc/shorewall/actions
+/etc/shorewall/arprules
+/etc/shorewall/blrules
+/etc/shorewall/clear
+/etc/shorewall/conntrack
+/etc/shorewall/ecn
+/etc/shorewall/findgw
+/etc/shorewall/hosts
+/etc/shorewall/init
+/etc/shorewall/initdone
+/etc/shorewall/interfaces
+/etc/shorewall/lib.private
+/etc/shorewall/maclist
+/etc/shorewall/mangle
+/etc/shorewall/nat
+/etc/shorewall/netmap
+/etc/shorewall/params
+/etc/shorewall/policy
+/etc/shorewall/providers
+/etc/shorewall/proxyarp
+/etc/shorewall/refresh
+/etc/shorewall/refreshed
+/etc/shorewall/restored
+/etc/shorewall/routes
+/etc/shorewall/rtrules
+/etc/shorewall/rules
+/etc/shorewall/scfilter
+/etc/shorewall/secmarks
+/etc/shorewall/shorewall.conf
+/etc/shorewall/snat
+/etc/shorewall/start
+/etc/shorewall/started
+/etc/shorewall/stop
+/etc/shorewall/stopped
+/etc/shorewall/stoppedrules
+/etc/shorewall/sysconfig
+/etc/shorewall/tcclasses
+/etc/shorewall/tcclear
+/etc/shorewall/tcdevices
+/etc/shorewall/tcfilters
+/etc/shorewall/tcinterfaces
+/etc/shorewall/tcpri
+/etc/shorewall/tunnels
+/etc/shorewall/zones
+endef
+
+define Package/shorewall/install
+	$(INSTALL_DIR) $(1)/etc/init.d/
+	$(INSTALL_DIR) $(1)/etc/shorewall/
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
+	$(INSTALL_DIR) $(1)/usr/sbin/
+	$(INSTALL_DIR) $(1)/usr/share/shorewall/
+	$(INSTALL_DIR) $(1)/usr/lib/shorewall/
+	$(CP) $(PKG_INSTALL_DIR)/etc/init.d/shorewall $(1)/etc/init.d/
+	$(CP) $(PKG_INSTALL_DIR)/etc/shorewall/. $(1)/etc/shorewall/
+	$(INSTALL_BIN) ./files/hotplug_iface $(1)/etc/hotplug.d/iface/05-shorewall
+	$(INSTALL_BIN) ./files/vardir $(1)/etc/shorewall/
+	$(INSTALL_BIN) ./files/hostname $(1)/etc/shorewall/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/shorewall/. $(1)/usr/lib/shorewall/
+	$(CP) $(PKG_INSTALL_DIR)/usr/share/shorewall/. $(1)/usr/share/shorewall/
+endef
```

Comment:
I could nitpick here about alphabetical order, but there are probably many packages that don't do it. Either way I prefer it.

Reply:
Done